File: leetcode-submissions/2425_Bitwise_XOR_of_All_Pairings.ts
function xorAllNums(nums1: number[], nums2: number[]): number {
    const map:Map<number,number>=new Map();
    for(const ele of nums1){
       
            map.set(ele,nums2.length);
        
    }
    for(const ele of nums2){
        
            map.set(ele,nums1.length);

    }
    let xor:number=0;
    for(const [element,frequency] of map.entries()){
        console.log(`${element}->${frequency}`);
        if(frequency%2){
            xor^=element;
        }
    }
    return xor;
};

File: leetcode-submissions/2425_Bitwise_XOR_of_All_Pairings.cpp
class Solution {
public:
    int xorAllNums(vector<int>& nums1, vector<int>& nums2) {
        int result=0;
        if(nums1.size()%2){
// if the nums1 is odd then we need to use all the elements of the nums2
    for(int ele :nums2){
        result^=ele;
    }
        }
        if(nums2.size()%2){
            for(int ele : nums1){
                result^=ele;
            }
        }
        return result;
    }
};